app-search-node.d.ts

```diff
@@ -0,0 +1,725 @@
+declare module "@elastic/app-search-node" {
+  export = AppSearchClient;
+
+  namespace AppSearchClient {
+    export interface DocumentResponse {
+      id: string;
+      errors: string[];
+    }
+
+    export interface SingleDocumentResponse {
+      id: string;
+    }
+    export interface Paging {
+      total_pages: number;
+      size: number;
+      current: number;
+      total_results: number;
+    }
+    export interface SearchResponse {
+      meta: {
+        warnings: string[];
+        page: Paging;
+        alerts: string[];
+        precision?: number;
+        request_id?: string;
+        engine?: {
+          name: string;
+          type: string;
+        };
+      };
+      results: Record<string, any>[];
+    }
+
+    export interface PagedResponse {
+      meta: {
+        page: Paging;
+      };
+      results: Record<string, any>[];
+    }
+
+    export interface QuerySuggestionsResponse {
+      meta: {
+        request_id: string;
+      };
+      results: {
+        documents: QuerySuggestion[];
+      };
+    }
+
+    export interface QuerySuggestion {
+      suggestion: string;
+    }
+
+    export interface DeleteResponse {
+      deleted: boolean;
+    }
+
+    export interface DocumentDeleteResponse {
+      deleted: boolean;
+      result: boolean;
+      id: string;
+    }
+  }
+
+  class AppSearchClient {
+    /**
+     * @example
+     *
+     * // Typical usage:
+     * const apiKey = 'private-mu75psc5egt9ppzuycnc2mc3'
+     * const baseUrlFn = () => 'http://localhost:3002/api/as/v1/'
+     * const client = new AppSearchClient(undefined, apiKey, baseUrlFn)
+     *
+     * // Deprecated usage for Swiftype.com:
+     * const hostIdentifier = "host-c5s2mj";
+     * const apiKey = "private-mu75psc5egt9ppzuycnc2mc3";
+     * const client = new AppSearchClient(hostIdentifier, apiKey)
+     */
+    constructor(
+      accountHostKey: string | undefined,
```

Comment:
Can this be `accountHostKey?: string`? Not sure if the compiler will complain about optional arguments preceding required arguments but wanted to raise.

Reply:
The compiler *does* complain about the optional arguments preceding required ones. Maybe with the upcoming 8.0 release we can change this signature to be better.